meson.build

```diff
@@ -0,0 +1,714 @@
+project('geany', 'c', 'cpp', version: '1.38')
+
+cc = meson.get_compiler('c')
+cxx = meson.get_compiler('cpp')
+prefix = get_option('prefix')
+
+deps_in = [
+	['glib-2.0',    '2.32'],
+	['gmodule-2.0', '2.32'],
+	['gtk+-3.0',    '3.0']
+]
+
+deps = []
+deps_for_pc = ''
+foreach dep : deps_in
+	deps += [dependency(dep[0], version: '>= ' + dep[1])]
+	deps_for_pc += ' ' + dep[0] + ' >= ' + dep[1]
+endforeach
+
+glib = deps[0]
+
+cdata = configuration_data()
+check_headers = [
+  'dlfcn.h',
+  'errno.h',
+  'fcntl.h',
+  'glob.h',
+  'inttypes.h',
+  'limits.h',
+  'locale.h',
+  'memory.h',
+  'stdint.h',
+  'stdlib.h',
+  'strings.h',
+  'string.h',
+  'stdbool.h',
+  'sys/stat.h',
+  'sys/time.h',
+  'sys/types.h',
+  'unistd.h',
+]
+
+check_functions = [
+  ['gnu_get_libc_version', '#include<gnu/libc-version.h>'],
+  ['fnmatch', '#include <fnmatch.h>'],
+  ['ftruncate', '#include <unistd.h>'],
+  ['mkstemp', '#include <stdlib.h>'],
+  ['regcomp', '#include <regex.h>'],
+  ['socket', '#include <sys/socket.h>'],
+  ['strerror', '#include <string.h>'],
+  ['strstr', '#include <string.h>'],
+]
+
+foreach h : check_headers
+  define = 'HAVE_' + h.underscorify().to_upper()
+  cdata.set(define, cc.has_header(h))
+endforeach
+
+foreach f : check_functions
+  define = 'HAVE_' + f.get(0).underscorify().to_upper()
+  cdata.set(define, cc.has_function(f.get(0), prefix : f.get(1)))
+endforeach
+
+cdata.set_quoted('PACKAGE', 'geany')
+cdata.set_quoted('PACKAGE_BUGREPORT', 'https://github.com/geany/geany/issues')
+cdata.set_quoted('PACKAGE_NAME','Geany')
+cdata.set_quoted('PACKAGE_STRING', 'Geany ' + meson.project_version())
+cdata.set_quoted('PACKAGE_VERSION', meson.project_version())
+cdata.set('GETTEXT_PACKAGE', 'PACKAGE')
+cdata.set('VERSION', meson.project_version())
+cdata.set('ENABLE_NLS', true)
+cdata.set('HAVE_VTE', get_option('vte'))
+cdata.set('HAVE_PLUGINS', get_option('plugins'))
+cdata.set('GEANY_DATA_DIR', join_paths(prefix, get_option('datadir'), 'geany'))
+cdata.set('GEANY_DOC_DIR', join_paths(prefix, get_option('datadir'), 'doc', 'geany'))
+cdata.set('top_srcdir', meson.source_root())
+cdata.set('top_builddir', meson.build_root())
+
+# for geany.pc (adapted from GTK+)
+pcconf = cdata
```

Comment:
There's [a function that will generate .pc files](https://mesonbuild.com/Pkgconfig-module.html#pkgconfig-module) more automagically...

Reply:
Thanks. I'll keep it for now as it seems to work, and is shared with autotools build.

Reply:
The automagic module is far less verbose than  `pcconf.set('DEPENDENCIES', deps_for_pc)` or redefining a variety of already existing variables.

Additionally, the automagic module will create a geany-uninstalled.pc file which can be used to compile third-party software against uninstalled development versions of geany, which may be useful.